core/src/main/java/org/apache/james/core/Username.java

```diff
@@ -31,7 +33,7 @@
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
 
-public class Username {
+public class Username implements Serializable {
```

Comment:
could you explain the changes done to Username class?

Reply:
That would be a breaking change, no?

Reply:
that a pity, but yes. Not sure how to handle that: it seems important to have a Username field in the User object, but I don't know what to suggest as a migration process.

Reply:
1. Can we agree that serializing a core model just backfired? Id rather not do the same mistake twice and introduce a dto for it.

2. We need a migration strategy. List user on old server, then create user on new server, implying password reset? This would need to be documented. Discussed as well?

Reply:
the User class use is not so widespread. I could maybe consider to keep it as is and replace it everywhere it is possible by a new one (so everywhere but where it is serialized)

Reply:
see my PR and its description here: https://github.com/linagora/james-project/pull/2842 . Maybe time to remove it else?